### ISSUE TITLE
Emacs 29 master declares global-linum-mode obsolete triggers backtrace with git-gutter load

### ISSUE THREAD

**Issue by mbarton98** [emacsorphanage/git-gutter#221]
git-gutter.el checks for global-linum-mode and after this commit in the Emacs master branch, that symbol is not defined.

```
fa37deba2eaca900fa3da6624c287ddec1fa61c4
Author:     Stefan Kangas <stefankangas@gmail.com>
AuthorDate: Wed Sep 21 14:22:26 2022 +0200
Commit:     Stefan Kangas <stefankangas@gmail.com>
CommitDate: Wed Sep 21 14:22:26 2022 +0200

Parent:     b0a85111e5 Add new function xdg-session-type to xdg.el
Contained:  master
Follows:    emacs-28.2 (159113)

; Explicitly declare linum-mode obsolete

* lisp/obsolete/linum.el (linum-mode, global-linum-mode):
Explicitly declare obsolete to warn the user, and recommend using
'display-line-numbers-mode' instead.
Suggested by Philip Kaludercic <philipk@posteo.net>.

1 file changed, 3 insertions(+)
lisp/obsolete/linum.el | 3 +++

modified   lisp/obsolete/linum.el
@@ -248,6 +248,9 @@ linum-unload-function
 (defconst linum-version "0.9x")
 (make-obsolete-variable 'linum-version 'emacs-version "28.1")
 
+(make-obsolete 'linum-mode #'display-line-numbers-mode "29.1")
+(make-obsolete 'global-linum-mode #'global-display-line-numbers-mode "29.1")
+
 (provide 'linum)
 
 ;;; linum.el ends here
```
Here is the backtrace.
```
Debugger entered--Lisp error: (void-variable global-linum-mode)
  byte-code("\10\203\15\0\301\302!\204\15\0\303 \210\301\207" [global-linum-mode boundp git-gutter-fringe git-gutter:linum-setup] 2)
  (global-git-gutter-mode 1)
  (condition-case err (global-git-gutter-mode 1) ((debug error) (funcall use-package--warning66 :init err)))
  (progn (condition-case err (global-git-gutter-mode 1) ((debug error) (funcall use-package--warning66 :init err))) (if (not (require 'git-gutter nil t)) (display-warning 'use-package (format "Cannot load %s" 'git-gutter) :error)))
  (condition-case err (progn (condition-case err (global-git-gutter-mode 1) ((debug error) (funcall use-package--warning66 :init err))) (if (not (require 'git-gutter nil t)) (display-warning 'use-package (format "Cannot load %s" 'git-gutter) :error))) ((debug error) (funcall use-package--warning66 :catch err)))
  load-with-code-conversion("/Users/bartm002/.emacs.d/myinit.el" "/Users/bartm002/.emacs.d/myinit.el" nil nil)
  load-file("/Users/bartm002/.emacs.d/myinit.el")
  org-babel-load-file("/Users/bartm002/.emacs.d/myinit.org")
  load-with-code-conversion("/Users/bartm002/.emacs.d/init.el" "/Users/bartm002/.emacs.d/init.el" t t)
  load("/Users/bartm002/.emacs.d/init" noerror nomessage)
  startup--load-user-init-file(#f(compiled-function () #<bytecode 0x15526e98caa805>) #f(compiled-function () #<bytecode -0x1f3c6feddc0d94b5>) t)
  command-line()
  normal-top-level()
```
For now I tried defining that symbol and things appear to be working in my usecase.
```
(use-package git-gutter
  :ensure t
  :init
  (setq global-linum-mode nil)
  (global-git-gutter-mode +1))
```

**Comment by jcs090218** [comment]
Should be resolved in #222.

**Comment by mbarton98** [comment]
Confirmed. That resolved this issue.
